engineering/documentation.md

```diff
@@ -0,0 +1,226 @@
+
+### Don't alienate the reader
+
+Avoid gendered pronouns. "He or she" should be rewritten as "they".
+There is no reason for your document to make assumptions on the reader's identity.
+
+Avoid sexist connotations.
+For instance, "mom-proof" implies mothers are not technical. Tell that to Grace Hopper.
+You can read more about sexist language and how to avoid it
+[here](https://www.hamilton.edu/academics/centers/writing/writing-resources/avoiding-sexist-language).
+
+Avoid ableist language. Avoid words like "crazy" or "insane".
+More on ableist language [here](https://www.copyediting.com/disabling-ableist-language/#.Wk6yeFQ-dE4).
+
+In general, if someone would find what you wrote offensive it's time to change it.
+There's no excuse to offend anyone in a technical document.
+
+### Take accessibility in mind
+
+It is probably impossible to write documentation that is fully accessible to every single person in the world.
+This doesn't mean we should not strive for documentation that is as accessible as possible.
+
+Many readers might have different degrees of visual impairments.
+While using images avoid attaching meaning to colors that are traditionally hard to distinguish.
+You can read more on what color combinations to avoid [here](http://blog.usabilla.com/how-to-design-for-color-blindness/).
+
+For more extreme cases the user will not be able to see the image at all.
+Take that into account and try to add captions explaining the image where possible.
+At least make usage of the `alt` attributes on HTML and markdown.
+
+## Documentation Artifacts
+
+Now that we've covered how to write, let's talk about what to write.
+This section lists the pieces of documentation we care about at source{d}.
+
+### README.md
+
+The `README.md` file in the root of your repository is the entrypoint to your project.
+At this point you still don't know whether the reader is a new-comer, an experienced user,
+or the most prolific contributor.
+
+Give an overview of the project goal and its current state.
+Is this a demo, a library, or maybe some tool?
+Should it be used in production? Is too soon, or maybe too late because it's been deprecated?
+
+For tools, it is also worth including basic installation instructions.
+If the instructions are long they should be part of a different document linked from this one.
+
+This document also serves as the index for all other documentation.
+`README.md` can also be found in folders other than the root of the repository.
+In that case it should document what the subtree below the folder contains.
+This is useful when you have libraries and binaries in a single repository.
+
+TL;DR:
+- Audience: Everyone.
+- Content: Description of the project and its current state. Links to other resources.
+- Example: https://github.com/twbs/bootstrap/blob/v4-dev/README.md
+
+### CONTRIBUTING.md
+
+This document describes what contributors should know in order to contribute to the project.
```

Comment:
since there's probably a lot of common practices, would it be useful to have a standard org-wide `CONTRIBUTING.md` just ready to copy and paste for all new projects?

Reply:
Nvm, we have it: https://github.com/src-d/guide/blob/master/engineering/documents/CONTRIBUTING.md. Still think it'd be useful to mention it here, though.

Reply:
I wonder if we should copy it around on every repo or simply link to it?
It seems a bit dangerous to simply copy things around in every repo tbh

What do you think, @mcuadros ?

Reply:
The contributing file has custom parts, maybe half of it, so you need to having it on the repo. You can link to a more regeneral stuff, but at the end what I saw in other orgs, is that just simply copy the file, since doesn't change to much during the time.

Reply:
I added a reference to that file and added also a `Template` field on the reference section below.